Tests/AzureRMAcceptanceTest.ps1

```diff
@@ -17,50 +17,64 @@ param (
     [string]$azureTenantId
 
 )
+# Note: Administrator PowerShell required to run this script.

```

Comment:
In the one of Nov release (11/8?), I remember there was a regression in Azure SDK.
"Microsoft.WindowsAzure.Storage.StorageException: Could not load file or assembly 'System.Spatial" while I do something like:
$a=Get-AzureRmStorageAccount
Get-AzureStorageTable -Context $a[0].Context

To avoid such in the future, would you consider to add a test case to cover it? thx
,

Reply:
@jianyunt Which version of the dll is it saying it can't load?

Reply:
@MiYanni In CloudShell we call Get-AzureStorageTable. The bug was in AzurePS

Repro:
 
$a=Get-AzureRmStorageAccount
Get-AzureStorageTable -Context $a[0].Context

In cloudshell,

PS Azure:\> dir -force -verbose
VERBOSE: Using TenantId '72f988bf-86f1-41af-91ab-2d7cd011db47' from 'ACC_TID' environment variable...
VERBOSE: Microsoft.WindowsAzure.Storage.StorageException: Could not load file or assembly 'System.Spatial, Version=5.8.1.0, Culture=neutral,
PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified. HTTP Status Code: 200 - HTTP Error
Message: OK ---> Microsoft.WindowsAzure.Storage.StorageException: Could not load file or assembly 'System.Spatial, Version=5.8.1.0,
Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified. --->
System.IO.FileNotFoundException: Could not load file or assembly 'System.Spatial, Version=5.8.1.0, Culture=neutral,
PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.
   at Microsoft.Data.OData.JsonLight.ODataJsonLightPropertyAndValueDeserializer.ReadPrimitiveValue(Boolean insideJsonObjectValue,
IEdmPrimitiveTypeReference expectedValueTypeReference, Boolean validateNullValue, String propertyName)
   at Microsoft.Data.OData.JsonLight.ODataJsonLightPropertyAndValueDeserializer.ReadNonEntityValueImplementation(String payloadTypeName,
IEdmTypeReference expectedTypeReference, DuplicatePropertyNamesChecker duplicatePropertyNamesChecker, CollectionWithoutExpectedTypeValidator
collectionValidator, Boolean validateNullValue, Boolean isTopLevelPropertyValue, Boolean insideComplexValue, String propertyName, Boolean
readRawValueEvenIfNoTypeFound)
   at Microsoft.Data.OData.JsonLight.ODataJsonLightEntryAndFeedDeserializer.ReadEntryDataProperty(IODataJsonLightReaderEntryState entryState,
IEdmProperty edmProperty, String propertyTypeName)
   at Microsoft.Data.OData.JsonLight.ODataJsonLightEntryAndFeedDeserializer.ReadEntryPropertyWithValue(IODataJsonLightReaderEntryState
entryState, String propertyName)
   at Microsoft.Data.OData.JsonLight.ODataJsonLightEntryAndFeedDeserializer.<>c__DisplayClass2.<ReadEntryContent>b__0(PropertyParsingResult
propertyParsingResult, String propertyName)
   at Microsoft.Data.OData.JsonLight.ODataJsonLightDeserializer.ProcessProperty(DuplicatePropertyNamesChecker duplicatePropertyNamesChecker,
Func`2 readPropertyAnnotationValue, Action`2 handleProperty)
   at Microsoft.Data.OData.JsonLight.ODataJsonLightEntryAndFeedDeserializer.ReadEntryContent(IODataJsonLightReaderEntryState entryState)
   at Microsoft.Data.OData.JsonLight.ODataJsonLightReader.ReadEntryStart(DuplicatePropertyNamesChecker duplicatePropertyNamesChecker,
SelectedPropertiesNode selectedProperties)
   at Microsoft.Data.OData.JsonLight.ODataJsonLightReader.ReadAtFeedStartImplementationSynchronously()
   at Microsoft.Data.OData.ODataReaderCore.ReadImplementation()
   at Microsoft.Data.OData.ODataReaderCore.InterceptException[T](Func`1 action)
   at
Microsoft.WindowsAzure.Storage.Table.Protocol.TableOperationHttpResponseParsers.TableQueryPostProcessGeneric[TElement,TQueryType](Stream
responseStream, Func`6 resolver, HttpWebResponse resp, TableRequestOptions options, OperationContext ctx, String accountName)
   at Microsoft.WindowsAzure.Storage.Table.TableQuery`1.<>c__DisplayClass10`2.<QueryImpl>b__f(RESTCommand`1 cmd, HttpWebResponse resp,
OperationContext ctx)
   at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.ProcessEndOfRequest[T](ExecutionState`1 executionState)
   at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.ExecuteSync[T](RESTCommand`1 cmd, IRetryPolicy policy, OperationContext
operationContext)
   --- End of inner exception stack trace ---
   at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.ExecuteSync[T](RESTCommand`1 cmd, IRetryPolicy policy, OperationContext
operationContext)
   at Microsoft.WindowsAzure.Storage.Table.TableQuery`1.<>c__DisplayClass7.<ExecuteInternal>b__6(IContinuationToken continuationToken)
   at Microsoft.WindowsAzure.Storage.Core.Util.CommonUtility.<LazyEnumerable>d__0`1.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at Microsoft.WindowsAzure.Commands.Storage.Table.Cmdlet.GetAzureStorageTableCommand.<ListTablesByName>d__12.MoveNext()
   at Microsoft.WindowsAzure.Commands.Storage.Table.Cmdlet.GetAzureStorageTableCommand.WriteTablesWithStorageContext(IEnumerable`1
tableList)
   at Microsoft.WindowsAzure.Commands.Storage.Table.Cmdlet.GetAzureStorageTableCommand.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
   --- End of inner exception stack trace ---
Request Information
RequestID:cd696af6-0002-00db-515f-5ade47000000
RequestDate:Fri, 10 Nov 2017 20:05:28 GMT
StatusMessage:OK

Reply:
@raghushantha I found the exact issue on StackOverflow: https://stackoverflow.com/questions/47211369/failing-to-get-azurestoragetable-due-to-missing-dependency-system-spatial

From what I'm told, this is caused because you are running 5.0.0 of the storage modules. If you update Azure.Storage to anything beyond 5.0.0, it'll fix this issue. It is recommended to simply get the latest version of AzureRm, which will have all of the updated modules/dependencies.

Reply:
I reported the issue, then your team made the fix, 5.0.1?, and we have to update our image before our release. By adding a test case will present to publish 5.0.0 to our gallery, right? :)

Reply:
@jianyunt From what I'm told, after 5.0.0 release, we added static analysis on check-in to validate that all the required DLLs are part of the module package. So, this issue would not happen again.